local-volume/provisioner/pkg/util/volume_util.go

```diff
@@ -94,6 +98,16 @@ func (u *volumeUtil) DeleteContents(fullPath string) error {
 	return nil
 }
 
+// GetFsAvailable returns available capacity about a mounted filesystem. fullPath
+// is the pathname of any file within the mounted filesystem.
+func (u *volumeUtil) GetFsAvailable(fullPath string) (uint64, error) {
+	var s syscall.Statfs_t
+	if err := syscall.Statfs(fullPath, &s); err != nil {
+		return 0, err
+	}
+	return uint64(s.Frsize) * s.Bavail, nil
```

Comment:
I found this [article](http://docs.oracle.com/cd/E19253-01/817-5093/fsfilesysappx-10/index.html) which explains this pretty well (the concept applies to most fs).

What is not quite clear is how this was handled in all sorts of filesystem; but it looks like the values are usually the same (the option to create different fragment size is actually removes the option from mke2fs, though man page still has the option).

Using Frsize here is mainly based on cadvisor https://github.com/google/cadvisor/blob/aee77359083ccef523d0299316e831f1aed712bd/fs/fs.go#L530

Also, df uses frsize as welll (fall back to bsize if it is not available): https://github.com/coreutils/gnulib/blob/2a0c08e48834ae056964922ee6d21ccd223d1d80/lib/fsusage.c#L131